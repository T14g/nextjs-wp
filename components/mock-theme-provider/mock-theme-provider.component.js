import React from "react";
import 'jest-styled-components'
import { ThemeProvider } from "styled-components";
import Theme from "../../themes/default";

export const MockThemeProvider = ({ children }) => (
    <ThemeProvider theme={Theme}>{children}</ThemeProvider>
)