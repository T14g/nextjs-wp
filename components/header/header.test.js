import React from "react";
import renderer from 'react-test-renderer';
import Header from "./header.component";
import { MockThemeProvider } from "../mock-theme-provider/mock-theme-provider.component";

const themeMock = {
    header: {
        background: '#fff',
        color: '#000'
    }
};

it('Render Header correctly', () => {
    const tree = renderer.create(
        <MockThemeProvider>
            <Header />
        </MockThemeProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
