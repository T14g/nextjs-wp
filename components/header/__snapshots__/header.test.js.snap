// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header correctly 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <a
      className=""
      href="/"
    >
      T14g.com
    </a>
    <ul
      className=""
    >
      <li
        className=""
      >
        Home
      </li>
      <li
        className=""
      >
        Blog
      </li>
      <li
        className=""
      >
        About
      </li>
    </ul>
  </div>
</div>
`;
